cmd/gitserver/server/server_test.go

```diff
@@ -971,6 +971,39 @@ func TestHandleRepoUpdate(t *testing.T) {
 		t.Fatal(diff)
 	}
 
+	// Create a new request with the Since interval to a long duration to verify that the
+	// update is skipped the next time this request is sent.
+	reqWithSince := protocol.RepoUpdateRequest{
+		Repo:  repoName,
+		Since: time.Duration(time.Hour * 1),
+	}
+
+	reqWithSince.Since = time.Duration(time.Hour * 1)
+	b, err := json.Marshal(reqWithSince)
+	if err != nil {
+		t.Fatal(err)
+	}
+	req = newRequest("GET", "/repo-update", bytes.NewReader(b))
+	s.handleRepoUpdate(rr, req)
+
+	expected, err := db.GitserverRepos().GetByID(ctx, dbRepo.ID)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// We will not ignore any fields in the comparison because we don't expect any changes
```

Comment:
alternative version suggestion:

```suggestion
	// We compare all fields because we don't expect any changes
```